### PR TITLE
Invibes Bid Adapter : infinite scroll tcf

### PR DESCRIPTION
**Pull request by gabrielIvbs**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

A new purpose was added.
Added a check for infinite scroll pages i.e pages that dynamically add content as the user scrolls.
Added a check for a placement being reused (multiple bids on the same placementId)
Changed the structure of the randomId to be a guid excluding '-'

### CODE REVIEW

File: modules/invibesBidAdapter.js [hunk 1]
```diff
@@ -87,10 +109,25 @@ function buildRequest(bidRequests, bidderRequest) {
...
+  }
 
   bidRequests.forEach(function (bidRequest) {
     bidRequest.startTime = new Date().getTime();
+
+    if (window.placementIds.includes(bidRequest.params.placementId)) {
+      isPlacementRefresh = true;
+    }
+
+    window.placementIds.push(bidRequest.params.placementId);
```
Comment (ncolletti) [inline modules/invibesBidAdapter.js:128]: Why is this data also being set to the window? What other code is reading it?

Comment (gabrielIvbs) [reply modules/invibesBidAdapter.js:128]: It's used for the case when the script is called multiple times, as the placementIds needs to be persistent.

File: modules/invibesBidAdapter.js [hunk 2]
```diff
@@ -87,10 +109,25 @@ function buildRequest(bidRequests, bidderRequest) {
   const _placementIds = [];
   const _adUnitCodes = [];
   let _customEndpoint, _userId, _domainId;
-  let _ivAuctionStart = bidderRequest.auctionStart || Date.now();
+  let _ivAuctionStart = Date.now();
+  if (window.placementIds == undefined) {
+    window.placementIds = [];
+  }
```
Comment (ncolletti) [inline modules/invibesBidAdapter.js:115]: Could this be updated to a more specific namespace?

Comment (ncolletti) [reply modules/invibesBidAdapter.js:115]: Suggest this is updated to remove the conditional check and always set either the existing value or set to an empty array:
`window.placementIds = window.placementIds || [];`

File: modules/invibesBidAdapter.js (hunk 1 above)
Comment (ncolletti) [reply modules/invibesBidAdapter.js:128]: Can you provide more details in your PR description as to the context of these changes, and provide the use cases your adapter is covering for publishers. Thank you

File: modules/invibesBidAdapter.js (hunk 2 above)
Comment (gabrielIvbs) [reply modules/invibesBidAdapter.js:115]: Thanks. I will do the change.

Comment (ncolletti) [reply modules/invibesBidAdapter.js]: Could you also update the namespace to be specific of your adapter?

```
window.invibes = window.invibes || {};
window.invibes.placementIds = window.invibes.placementIds || [];
```

File: modules/invibesBidAdapter.js [hunk 3]
```diff
@@ -49,14 +49,36 @@ registerBidder(spec);
...
+let isInfiniteScrollPage = false;
+let isPlacementRefresh = false;
 let _customUserSync;
 let _disableUserSyncs;
 
+function updateInfiniteScrollFlag() {
+  const { scrollHeight } = document.documentElement;
+
+  if (window.originalURL === undefined) {
+    window.originalURL = window.location.href;
```
Comment (ncolletti) [inline modules/invibesBidAdapter.js:66]: please nest these within your new window.invibes object

File: modules/invibesBidAdapter.js [hunk 4]
```diff
@@ -87,10 +109,23 @@ function buildRequest(bidRequests, bidderRequest) {
   const _placementIds = [];
   const _adUnitCodes = [];
   let _customEndpoint, _userId, _domainId;
-  let _ivAuctionStart = bidderRequest.auctionStart || Date.now();
+  let _ivAuctionStart = Date.now();
+  window.placementIds = window.placementIds || [];
```
Comment (ncolletti) [inline modules/invibesBidAdapter.js:113]: let's put `placementIds` on the `window.invibes` object

**Review by ncolletti — APPROVED** [review]
thanks for the adjustments, looks good

**Pull request merged**